### PR TITLE
[MU4] Accessibility. Master palette

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #9705 #9704

### CODE REVIEW

File: src/framework/accessibility/iaccessibleinterfaceregister.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace mu::accessibility {
+class IAccessibleInterfaceRegister : MODULE_EXPORT_INTERFACE
+{
+    INTERFACE_ID(IAccessibleInterfaceRegister)
+
+public:
+    virtual ~IAccessibleInterfaceRegister() = default;
+
+    using InterfaceGetter = std::function<QAccessibleInterface* (QObject*)>;
```
Comment (igorkorsukov) [inline src/framework/accessibility/iaccessibleinterfaceregister.h:37]: Do we really need this interface?
I believe that there is no need for Qt interfaces to appear in our interfaces.
It is better if we make an implimentation from IAccessible who will be a proxy for QAccessibleInterface

Or at least you need to rename IAccessibleInterfaceRegister to IQAccessibleInterfaceRegister

File: src/framework/accessibility/internal/accessibilitycontroller.cpp [hunk 2]
```diff
@@ -47,20 +47,21 @@ AccessibilityController::~AccessibilityController()
...
+    return static_cast<QAccessibleInterface*>(new AccessibleItemInterface(s_rootObject));
+}
+
 static QAccessibleInterface* muAccessibleFactory(const QString& classname, QObject* object)
 {
-#ifdef Q_OS_MAC
-    if (classname == QLatin1String("QQuickWindow")) {
-        return static_cast<QAccessibleInterface*>(new AccessibleItemInterface(s_rootObject));
+    auto accessibleInterfaceRegister = mu::modularity::ioc()->resolve<IAccessibleInterfaceRegister>("accessibility");
+    if (!accessibleInterfaceRegister) {
```
Comment (igorkorsukov) [inline src/framework/accessibility/internal/accessibilitycontroller.cpp:58]: This function is called very often, and `ioc()->resolve` is quite an overhead call, it is better to do it once

File: src/framework/accessibility/iaccessibleinterfaceregister.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    virtual void registerInterfaceGetter(const std::string& className, const InterfaceGetter& interfaceGetter) = 0;
```
Comment (igorkorsukov) [inline src/framework/accessibility/iaccessibleinterfaceregister.h:39]: if it uses a Qt type that cannot be gotten rid of, then it makes no sense to use `std::string`

File: src/framework/accessibility/iaccessibleinterfaceregister.h (hunk 1 above)
Comment (Eism) [reply src/framework/accessibility/iaccessibleinterfaceregister.h]: thanks, renamed

File: src/framework/accessibility/internal/accessibilitycontroller.cpp (hunk 2 above)
Comment (Eism) [reply src/framework/accessibility/internal/accessibilitycontroller.cpp:58]: fixed

File: src/framework/accessibility/iaccessibleinterfaceregister.h (hunk 3 above)
Comment (Eism) [reply src/framework/accessibility/iaccessibleinterfaceregister.h]: fixed

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**